### PR TITLE
fix parallel datamigrate pod anti affinity for the launcher(job) pod

### PR DESCRIPTION
**Pull request by xliuqq**
### Ⅰ. Describe what this PR does
The launcher (job pod) is used as a worker, so it should add **PodAntiAffinity** to prefer to not on the same node as worker pods.

### Ⅱ. Does this pull request fix one issue?

fixes #XXXX

### Ⅴ. Special notes for reviews

### CODE REVIEW

File: pkg/ddc/juicefs/data_migrate.go [hunk 1]
```diff
@@ -215,6 +220,46 @@ func (j *JuiceFSEngine) generateDataMigrateValueFile(r cruntime.ReconcileRequest
...
+		dataMigrateInfo.Affinity.PodAntiAffinity = &corev1.PodAntiAffinity{
+			PreferredDuringSchedulingIgnoredDuringExecution: []corev1.WeightedPodAffinityTerm{
+				podAffinityTerm,
+			},
+		}
+		return
+	}
+	// PodAntiAffinity not nil
+	dataMigrateInfo.Affinity.PodAntiAffinity.PreferredDuringSchedulingIgnoredDuringExecution =
+		append(dataMigrateInfo.Affinity.PodAntiAffinity.PreferredDuringSchedulingIgnoredDuringExecution, podAffinityTerm)
```
Comment (TrafalgarZZZ) [inline pkg/ddc/juicefs/data_migrate.go:260]: `append` can be applied to nil slice, so I think we can simplify this part with:
```
if dataMigrateInfo.Affinity.PodAntiAffinity == nil {
    dataMigrateInfo.Affinity.PodAntiAffinity = &corev1.PodAntiAffinity{}
}

dataMigrateInfo.Affinity.PodAntiAffinity.PreferredDuringSchedulingIgnoredDuringExecution =
		append(dataMigrateInfo.Affinity.PodAntiAffinity.PreferredDuringSchedulingIgnoredDuringExecution, podAffinityTerm)
```

File: charts/fluid-datamigrate/juicefs/templates/statefulset.yaml [hunk 2]
```diff
@@ -64,16 +65,16 @@ spec:
...
           preferredDuringSchedulingIgnoredDuringExecution:
             - podAffinityTerm:
                 labelSelector:
                   matchExpressions:
-                    - key: app
+                    - key: fluid.io/operation
                       operator: In
                       values:
-                        - {{ printf "%s-workers" .Release.Name }}
+                        - migrate-{{ .Release.Namespace }}-{{ .Release.Name }}
```
Comment (TrafalgarZZZ) [inline charts/fluid-datamigrate/juicefs/templates/statefulset.yaml:77]: Why don't we reuse `{{ .Values.datamigrate.affinity }}` just like what we do in job.yaml?

File: pkg/ddc/juicefs/data_migrate.go (hunk 1 above)
Comment (xliuqq) [reply pkg/ddc/juicefs/data_migrate.go]: fixed.

File: charts/fluid-datamigrate/juicefs/templates/statefulset.yaml (hunk 2 above)
Comment (xliuqq) [reply charts/fluid-datamigrate/juicefs/templates/statefulset.yaml]: fixed.

**Review by cheyang — APPROVED** [review]
/lgtm
/approve

**Pull request merged**